### PR TITLE
feat: Add missing eligible to join tag in CommunityMembershipSetupDialog

### LINKED ISSUE #14088: Requirements check in `JoinCommunityView` are using selected addresses instead of all
### Description

1. Initially, `JoinCommunityView` page uses ALL wallet accounts a user can have to determine if requirements are met.
2. After going through the `ShareAddresses` flow, and unselecting some accounts, then closing the popup, `JoinCommunityView` uses these new accounts selection to determine if requirements are met so it can cause an undesired situation where user is no longer able to go through the _reveal your addresses flow_.

Here a video for better understanding:

https://github.com/status-im/status-desktop/assets/97019400/bf994632-5715-4b9e-84dc-e242b1e86c36

**Summary:** `JoinCommunityView` page should allways consider ALL wallet accounts a user has.

### LINKED ISSUE #14171: Share addresses button enabled when selected account does not meet permissions
# Bug Report

## Description

1. When joining a community it is possible to select an account to share that does not meet the requirements for the community.
2. The share account to join button remains enabled in this case
3. Clicking it returns the user to the permissions overlay screen but they can no longer change what account is shared.
4. If the community is closed and then revisited the user can select another account to share

image

image

#### Expected behavior

User cannot proceed to share address when account does not meet requirements
OR
User can select a different account to share after returning to the permissions overlay screen

### Additional Information

- Status desktop version: 2.28 RC2 C26110
- Operating System: Mac

### PR DESCRIPTION
**Pull request by caybro**
### What does the PR do

- implement the eligibility check in C++, returning the highest possible role the user would be allowed to join under
- enable/disable the "Share" button based on the above permissions check
- remove all the locally placed components, access teh popup only via Global/Popups
- calculate the `accessType` internally based on the permissions present
- update the eligibility as the async check for permissions is finished
- fix the permissions panel background color
- partially revert the share/finish/cancel buttons behavior; it must be one button due to StatusStackModal limitations
- fix some other minor UI issues or differences to current Figma designs
- adjust SB, add the possibility to play around with different permission models

Fixes https://github.com/status-im/status-desktop/issues/14100
Fixes #14171
Fixes #14088

### Affected areas

CommunityMembershipSetupDialog, SharedAddressesPanel, SharedAddressesPermissionsPanel

### Screenshot of functionality (including design for comparison)

- [x] I've checked the design and this PR matches it

image

image

### CODE REVIEW

File: src/app/modules/main/chat_section/module.nim [hunk 1]
```diff
@@ -388,11 +388,12 @@ proc rebuildCommunityTokenPermissionsModel(self: Module) =
...
 proc reevaluateRequiresTokenPermissionToJoin(self: Module) =
   let community = self.controller.getMyCommunity()
-  var hasBecomeMemberOrBecomeAdminPermissions = false
-  for id, tokenPermission in community.tokenPermissions:
-    if tokenPermission.`type` == TokenPermissionType.BecomeMember or tokenPermission.`type` == TokenPermissionType.BecomeAdmin:
-      hasBecomeMemberOrBecomeAdminPermissions = true
-  self.view.setRequiresTokenPermissionToJoin(hasBecomeMemberOrBecomeAdminPermissions)
+  var joinPermissionsChanged = false
+  for _, tokenPermission in community.tokenPermissions:
+    if tokenPermission.`type` == TokenPermissionType.BecomeMember or tokenPermission.`type` == TokenPermissionType.BecomeAdmin or tokenPermission.`type` == TokenPermissionType.BecomeTokenMaster:
```
Comment (caybro) [inline src/app/modules/main/chat_section/module.nim:393]: Check this out @endulab @jrainville pls

Comment (caybro) [reply src/app/modules/main/chat_section/module.nim:393]: @mprakhov too pls :)

File: ui/app/AppLayouts/Communities/controls/CommunityEligibilityTag.qml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+import StatusQ.Core.Theme 0.1
+
+import AppLayouts.Communities.helpers 1.0
+
+import utils 1.0
+
+Rectangle {
+    id: root
+
+    required property int /*PermissionTypes.Type*/ eligibleToJoinAs
```
Comment (jrainville) [inline ui/app/AppLayouts/Communities/controls/CommunityEligibilityTag.qml:14]: Leftover?

File: ui/app/AppLayouts/Communities/panels/SharedAddressesPanel.qml [hunk 3]
```diff
@@ -31,6 +31,7 @@ Control {
     required property int totalNumOfAddressesForSharing
     required property bool profileProvesOwnershipOfSelectedAddresses
     required property bool allAddressesToRevealBelongToSingleNonProfileKeypair
+    required property int /*PermissionTypes.Type*/ eligibleToJoinAs
```
Comment (jrainville) [inline ui/app/AppLayouts/Communities/panels/SharedAddressesPanel.qml:34]: leftover?

File: ui/imports/shared/popups/CommunityMembershipSetupDialog.qml [hunk 4]
```diff
@@ -159,18 +168,25 @@ StatusStackModal {
...
+                        return walletType === Constants.generatedWalletType
                     }
 
                     expression: {
-                        return isGenerated(modelLeft)
+                        return isGenerated(modelLeft.walletType)
                     }
+                    expectedRoles: ["walletType"]
                 },
+                // FIXME sort by balances
```
Comment (jrainville) [inline ui/imports/shared/popups/CommunityMembershipSetupDialog.qml:189]: Still needed?

**Review by jrainville — COMMENTED** [review]
LGTM

File: ui/imports/shared/popups/CommunityMembershipSetupDialog.qml (hunk 4 above)
Comment (caybro) [reply ui/imports/shared/popups/CommunityMembershipSetupDialog.qml:189]: Yeah, the sorting will be different here (followup PRs)

File: src/app/modules/main/chat_section/module.nim (hunk 1 above)
Comment (caybro) [reply src/app/modules/main/chat_section/module.nim:393]: @mprakhov can you create/test that scenario? I was under the impression that `type == TokenPermissionType.BecomeTokenMaster` also qualifies as a "join" permission; if not, I can remove the condition here

**Review by jrainville — APPROVED** [review]

**Pull request merged**